### PR TITLE
Sidecar injection by reconciliation

### PR DESCRIPTION
**Pull request by danielpcox**
[sc-22489]

Companion PR to https://github.com/greymatter-io/operator/pull/151

### CODE REVIEW

**Review by alecholmez — CHANGES_REQUESTED** [review]
Just some minor things

File: inputs.cue [hunk 1]
```diff
@@ -45,10 +45,17 @@ defaults: {
 	image_pull_secret_name: string | *"gm-docker-secret"
 	image_pull_policy:      corev1.#enumPullPolicy | *corev1.#PullAlways
 	xds_host:               "controlensemble.\(mesh.spec.install_namespace).svc.cluster.local"
+	sidecar_list:           [...string] | *["dashboard", "catalog", "controlensemble", "edge"]
+	proxy_port_name:        "proxy"
```
[the remark this answers is not in the document]
Comment (danielpcox) [reply inputs.cue:49]: Good plan, I'll add a comment.

File: inputs.cue [hunk 2]
```diff
@@ -45,10 +45,17 @@ defaults: {
...
 	redis_cluster_name:     "redis"
 	redis_host:             "\(redis_cluster_name).\(mesh.spec.install_namespace).svc.cluster.local"
-	spire_selinux_context:  string | *"s0:c30,c5"
-	sidecar_list:           [...string] | *["dashboard", "catalog", "controlensemble", "edge"]
+	redis_port:             6379
+	redis_db:               0
+	redis_username:         ""
+	redis_password:         ""
+	gitops_state_key_gm:      "gmHashes"  // the key in Redis for Grey Matter applied-state backups
+  gitops_state_key_k8s:     "k8sHashes" // the key in Redis for K8s applied-state backups
```
[the remark this answers is not in the document]
Comment (danielpcox) [reply inputs.cue:57]: K

[the remark this answers is not in the document]
Comment (danielpcox) [reply inputs.cue:57]: It doesn't actually look like this in my editor, if you're referring to the `gitops_state_key_k8s` key. I suspect this is just diff weirdness. If you're referring to the alignment of all the rest of the values, I've fixed that now. 👌

**Review by alecholmez — DISMISSED** [review]
This looks good to me now

**Review by alecholmez — APPROVED** [review]

**Pull request merged**